From c7d3d61f81ebbb5bc13c1b7ece1d0d39fa2e9224 Mon Sep 17 00:00:00 2001
From: Katelyn Gigante <clockwork.singularity@gmail.com>
Date: Mon, 2 Nov 2020 23:42:11 +1100
Subject: [PATCH] Ensure that GTK is linked on linux

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 552bb9bb..af465317 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,9 @@ if( NOT HUNSPELL_FOUND )
 		message("-- Found Hunspell at ${HUNSPELL_LIBRARIES}")
 	endif()
 endif()
+if (NOT WIN32 AND NOT APPLE)
+       pkg_check_modules(GTK3 REQUIRED gtk+-3.0)
+endif()
 
 include_directories("${PROJECT_BINARY_DIR}/src")
 include_directories("${PROJECT_SOURCE_DIR}/src")
@@ -36,6 +39,11 @@ target_link_libraries(${PROJECT_NAME} ${wxWidgets_LIBRARIES})
 target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
 target_link_libraries(${PROJECT_NAME} ${HUNSPELL_LIBRARIES})
 
+if (NOT WIN32 AND NOT APPLE)
+       link_directories(${GTK3_LIBRARY_DIRS})
+       target_link_libraries(${PROJECT_NAME} ${GTK3_LIBRARIES})
+endif()
+
 file(GLOB_RECURSE sources src/*.cpp)
 list(FILTER sources EXCLUDE REGEX win32_cli_wrapper.cpp)
 target_sources(magicseteditor PRIVATE ${sources})
-- 
2.28.0.windows.1

